pkg/cmd/repo/create/create.go

```diff
@@ -855,7 +855,11 @@ func interactiveRepoInfo(client *http.Client, hostname string, prompter iprompte
 		return "", "", "", err
 	}
 
-	visibilityOptions := []string{"Public", "Private", "Internal"}
+	visibilityOptions := []string{"Public", "Private"}
+	// orgs can also create internal repos
+	if owner != "" {
+		visibilityOptions = append(visibilityOptions, "Internal")
+	}
```

Comment:
This change makes sense after reviewing `interactiveRepoNameAndOwner` will return a non-empty string for owner if an organization:

https://github.com/cli/cli/blob/74b9724c02876b024d03f4c26680ed0ed79860ac/pkg/cmd/repo/create/create.go#L903-L908

Outside the scope of this issue, there are some types of organizations that cannot create `public` repositories like [Enterprise Managed Users](https://docs.github.com/en/enterprise-cloud@latest/admin/managing-iam/understanding-iam-for-enterprises/about-enterprise-managed-users); only noting a similar use case which would require inspection of the selected owner to detect.

> Managed user accounts cannot create public content or collaborate outside your enterprise. See "[Abilities and restrictions of managed user accounts](https://docs.github.com/en/enterprise-cloud@latest/admin/identity-and-access-management/understanding-iam-for-enterprises/abilities-and-restrictions-of-managed-user-accounts)."